python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -257,8 +229,7 @@ def __init__(
             # different nodes.
             prev_reader_node = None
             prev_reader = None
-            for reader in readers:
-                node = _get_reader_node_id(self, reader)
+            for reader, node in reader_to_node:
```

Comment:
Does the logic change anywhere, or passing in `reader_to_node` is just refactoring?

Reply:
You can read "Change 2" in the PR description for more details.